besu-plugins/linea-sequencer/sequencer/src/main/java/net/consensys/linea/sequencer/liveness/LivenessPlugin.java

```diff
@@ -0,0 +1,429 @@
+
+    if (scheduler != null) {
+      scheduler.shutdown();
+      try {
+        if (!scheduler.awaitTermination(5, TimeUnit.SECONDS)) {
+          scheduler.shutdownNow();
+        }
+      } catch (InterruptedException e) {
+        log.error("Error terminating scheduler: {}", e.getMessage(), e);
+        scheduler.shutdownNow();
+        Thread.currentThread().interrupt();
+      }
+    }
+
+    if (web3j != null) {
+      web3j.shutdown();
+    }
+
+    super.stop();
+
+    log.info("{} stopped", PLUGIN_NAME);
+  }
+
+  /**
+   * Gets the plugin configuration.
+   *
+   * @return The LivenessPluginConfiguration
+   */
+  private LivenessPluginConfiguration getConfiguration() {
+    return livenessPluginConfiguration();
+  }
+
+  @Override
+  public void onBlockAdded(AddedBlockContext addedBlockContext) {
+    if (!isPluginEnabled) return;
+
+    BlockHeader newBlock = addedBlockContext.getBlockHeader();
+    lastProcessedBlock.set(newBlock);
+
+    // Reset the last reported timestamp when a new block is added
+    lastReportedTimestamp.set(newBlock.getTimestamp());
+
+    log.debug(
+        "New block added: lastProcessedBlock ={}, lastReportedTimestamp ={}",
+        lastProcessedBlock.get().getNumber(),
+        lastReportedTimestamp.get());
+  }
+
+  /**
+   * Checks the timestamp of the last block and reports downtime if necessary. This method is called
+   * periodically by the scheduler.
+   */
+  private void checkBlockTimestampAndReport() {
+    if (!isPluginEnabled) return;
+
+    try {
+      BlockHeader lastBlock = lastProcessedBlock.get();
+      if (lastBlock == null) {
+        log.warn("No blocks available in the blockchain");
+        return;
+      }
+
+      long currentTimestamp = Instant.now().getEpochSecond();
+      long lastBlockTimestamp = lastBlock.getTimestamp();
+      long timeSinceLastBlock = currentTimestamp - lastBlockTimestamp;
+
+      log.debug(
+          "Checking block timestamp: lastBlockNumber={}, lastBlockTimestamp={}, currentTimestamp={}, timeSinceLastBlock={}s",
+          lastBlock.getNumber(),
+          lastBlockTimestamp,
+          currentTimestamp,
+          timeSinceLastBlock);
+
+      // Check if we need to report downtime
+      if (timeSinceLastBlock > livenessPluginConfiguration.maxBlockAgeMilliseconds()) {
+        // Only report if we haven't reported recently or if significant time has passed
+        long timeSinceLastReport = currentTimestamp - lastReportedTimestamp.get();
+        if (timeSinceLastReport > livenessPluginConfiguration.maxBlockAgeMilliseconds()) {
+          log.info(
+              "Sequencer appears to have been down: lastBlockNumber={}, lastBlockTimestamp={}, timeSinceLastBlock={}s",
+              lastBlock.getNumber(),
+              lastBlockTimestamp,
+              timeSinceLastBlock);
+
+          sendSequencerUptimeTransaction(lastBlockTimestamp, currentTimestamp);
+          lastReportedTimestamp.set(currentTimestamp);
```

Comment:
Leaving an observation that here we update `lastReportedTimestamp` with the system clock from ` Instant.now().getEpochSecond()`, in other places we update that it with the block timestamp